### PR TITLE
#14 - Incomplete documentation of Prefixed class

### PR DESCRIPTION
**Pull request by maurezen**
added javadoc to Prefixed

### CODE REVIEW

**Comment by dmarkov** [comment]
Thanks, let me find someone who can review this pull request

**Comment by dmarkov** [comment]
@krzyk please review

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 1]
```diff
@@ -108,7 +108,29 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
...
+     *
+     *
+     * <p>Basically this class is involved to cut off ocket keys of underlying
+     * bucket by some string known as prefix. If key is not started
+     * with prefix, it will be omitted
+     *
+     * <p>Example of usage:
+     * <pre>
+     * final Region region = new MkRegion(
+     * new TemporaryFolder().newFolder()
```
Comment (krzyk) [inline src/main/java/com/jcabi/s3/Bucket.java]: let's format this code (as it is in pre), so here e.g. indentation

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 2]
```diff
@@ -108,7 +108,29 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
         String pfx) throws IOException;
 
     /**
-     * Prefixed.
+     * Creates bucket with specified origin bucket and prefix.
+     *
```
Comment (krzyk) [inline src/main/java/com/jcabi/s3/Bucket.java]: please remove this line

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 3]
```diff
@@ -108,7 +108,29 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
...
+     * <pre>
+     * final Region region = new MkRegion(
+     * new TemporaryFolder().newFolder()
+     * );
+     * final Bucket bucket = region.bucket("test");
+     * new Ocket.Text(bucket.ocket("a/first.txt")).write("");
+     * new Ocket.Text(bucket.ocket("a/b/hello.txt")).write("");
+     * new Ocket.Text(bucket.ocket("a/b/f/2.txt")).write("");
+     * Bucket.Prefixed prefixed = new Bucket.Prefixed(
+     * bucket, "a/b/"
```
Comment (krzyk) [inline src/main/java/com/jcabi/s3/Bucket.java]: this should be indented

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 4]
```diff
@@ -108,7 +108,29 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
...
+     * );
+     * Iterable<String> list = prefixed.list(
+     * ""
```
Comment (krzyk) [inline src/main/java/com/jcabi/s3/Bucket.java]: here also

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 5]
```diff
@@ -108,7 +108,29 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
         String pfx) throws IOException;
 
     /**
-     * Prefixed.
+     * Creates bucket with specified origin bucket and prefix.
+     *
+     *
+     * <p>Basically this class is involved to cut off ocket keys of underlying
```
Comment (krzyk) [inline src/main/java/com/jcabi/s3/Bucket.java]: I think it will be better with such change: *... involved to cut off ...* -> *... is used to cut off ...*

**Comment by krzyk** [comment]
@maurezen some comments above

**Comment by maurezen** [comment]
@krzyk please take a look.

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 6]
```diff
@@ -108,7 +108,26 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
         String pfx) throws IOException;
 
     /**
-     * Prefixed.
```
Comment (krzyk) [inline src/main/java/com/jcabi/s3/Bucket.java]: You removed too much :)
Please leave *Creates bucket with specified origin bucket and prefix.* and a single empty comment line after it.

**Comment by krzyk** [comment]
@maurezen just a correction of my comment above :)

**Comment by maurezen** [comment]
@krzyk done =)

**Comment by krzyk** [comment]
@rultor merge pls

File: src/main/java/com/jcabi/s3/Bucket.java [hunk 7]
```diff
@@ -108,7 +108,28 @@ void remove(@NotNull(message = "S3 key can't be NULL") String key)
...
+     *     final Bucket bucket = region.bucket("test");
+     *     new Ocket.Text(bucket.ocket("a/first.txt")).write("");
+     *     new Ocket.Text(bucket.ocket("a/b/hello.txt")).write("");
+     *     new Ocket.Text(bucket.ocket("a/b/f/2.txt")).write("");
+     *     Bucket.Prefixed prefixed = new Bucket.Prefixed(
+     *         bucket, "a/b/"
+     *     );
+     *     Iterable<String> list = prefixed.list(
+     *         ""
+     *     ); // contains "hello.txt" and "f/2.txt"
```
Comment (yegor256) [inline src/main/java/com/jcabi/s3/Bucket.java]: what's the point of this indentation? see how javadocs are formatted, for example here: https://github.com/jcabi/jcabi-s3/blob/master/src/main/java/com/jcabi/s3/Ocket.java#L49-L62

**Comment by yegor256** [comment]
@maurezen one comment from me

**Comment by maurezen** [comment]
@yegor256 right, formatted the same way.

**Comment by yegor256** [comment]
@rultor merge

**Comment by dmarkov** [comment]
@krzyk Much obliged! I've added **15 mins** to your account in payment "51519704"

added +15 to your rating, now it is equal to [+3857](http://www.netbout.com/b/35035?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy pls

**Pull request merged**